15-greenhouse-monitor/firmware/greenhouse/greenhouse.ino

```diff
@@ -0,0 +1,628 @@
+// Copyright 2023 Blues Inc.  All rights reserved.
+// Use of this source code is governed by licenses granted by the
+// copyright holder including that found in the LICENSE file.
+
+#include <Arduino.h>
+#include <string.h>
+#include "Notecard.h"
+#include "Adafruit_BME280.h"
+#include "Adafruit_seesaw.h"
+#include "greenhouse.h"
+
+#define LIGHT_SENSOR_PIN (A1)
+#define DATA_FILE_NOTIFY "notify.qo"
+#define DATA_FILE_MONITOR "greenhouse.qo"
+#define DATA_FILE_ALERT "alert.qo"
+
+#define DATA_FIELD_APP "app"
+#define DATA_FIELD_ALERT_SEQUENCE "alert_seq"
+#define DATA_FIELD_ALERT_LEVEL "alert"
+#define DATA_FIELD_AIR_TEMP "air_temp"
+#define DATA_FIELD_AIR_HUMIDITY "air_humidity"
+#define DATA_FIELD_AIR_PRESSURE "air_pressure"
+#define DATA_FIELD_LIGHT_LEVEL "light_level"
+#define DATA_FIELD_SOIL_TEMP "soil_temp"
+#define DATA_FIELD_SOIL_MOISTURE "soil_moisture"
+
+#ifndef PRODUCT_UID
+#define PRODUCT_UID ""		// "com.my-company.my-name:my-project"
+#pragma message "PRODUCT_UID is not defined in this example. Please ensure your Notecard has a product identifier set before running this example or define it in code here. More details at https://bit.ly/product-uid"
+#endif
+
+// When set to true, all monitoring notes are synced immediately to Notehub.
+// When set to false, only alerts are synched immediately.
+#ifndef SYNC_MONITORING_NOTES
+#define SYNC_MONITORING_NOTES        (false)
+#endif
+
+#ifndef DEFAULT_REPORT_INTERVAL
+#define DEFAULT_REPORT_INTERVAL (1000*60*5)
+#endif
+
+#ifndef DEFAULT_POLL_ENVIRONMENT_INTERVAL
+#define DEFAULT_POLL_ENVIRONMENT_INTERVAL (1000*60*5)
+#endif
+
+#ifndef DEFAULT_POLL_SENSORS_INTERVAL
+#define DEFAULT_POLL_SENSORS_INTERVAL (1000*15)
+#endif
+
+#ifndef APP_NAME
+#define APP_NAME    "nf15"
+#endif
+
+
+void outputInterval(Stream& out, float vmin, float vmax) {
+    out.print('[');
+    if (isValueSet(vmin)) out.print(vmin);
+    out.print(',');
+    if (isValueSet(vmax)) out.print(vmax);
+    out.print(']');
+}
+
+inline bool isEmptyInterval(float vmin, float vmax) {
+    return vmin==vmax || (!isValueSet(vmin) && !isValueSet(vmax));
+}
+
+void outputThresholdAlert(Stream& out, const ThresholdAlert* a) {
+    out.print(alertLevelString(a->alertLevel));
+    if (a->threshold!=THRESHOLD_NONE) {
+        out.print(": threshold ");
+        out.print(thresholdString(a->threshold));
+    }
+}
+
+
+#define debug Serial
+
+Notecard notecard;
+
+Adafruit_BME280 bme280;
+bool bme280Initialized;
+bool bme280Messaged;
+float air_temp;
```

Comment:
Is there a distinction between snake_case variables and camelCase ones?  Not important, but I was just wondering since both are used here.

Reply:
The environment variables, event properties use snake_case, so wanted to keep those variables distinct. But not something I'm married to, can change for consistency.